src/script/event/EventRepository.js

```diff
@@ -598,21 +602,25 @@ export class EventRepository {
    * Handle a single event from the notification stream or WebSocket.
    *
    * @private
-   * @param {JSON} event - Backend event extracted from notification stream
+   * @param {JSON} event - Event coming from backend
    * @param {EventRepository.SOURCE} source - Source of event
-   * @returns {Promise} Resolves with the saved record or `true` if the event was skipped
+   * @returns {Promise} Resolves with the saved record or the plain event if the event was skipped
    */
   _handleEvent(event, source) {
-    return this._handleEventValidation(event, source)
-      .then(validatedEvent => this.processEvent(validatedEvent, source))
-      .catch(error => {
-        const isIgnoredError = EventRepository.CONFIG.IGNORED_ERRORS.includes(error.type);
-        if (!isIgnoredError) {
-          throw error;
-        }
-
+    const logObject = {eventJson: JSON.stringify(event), eventObject: event};
+    const validationResult = handleEventValidation(event, source, this.lastEventDate());
+    switch (validationResult) {
+      default:
         return event;
-      });
+      case EventValidation.IGNORED_TYPE:
+        this.logger.info(`Ignored event type '${event.type}'`, logObject);
+        return event;
```

Comment:
Wrong return type here, not a promise

Reply:
But we removed the `async`, so it is not a Promise anymore.